tap_zendesk/streams.py

```diff
@@ -554,6 +555,38 @@ def sync(self, state): # pylint: disable=unused-argument
         for policy in self.client.sla_policies():
             yield (self.stream, policy)
 
+
+class Calls(Stream):
+    name = "calls"
+    replication_method = "INCREMENTAL"
+
+    def sync(self, state):
+        # The incremental Talk endpoint isn't currently supported by the Zenpy
+        # library, though there is an open PR for getting that in there:
+        # https://github.com/facetoe/zenpy/pull/454
+        # If/when that gets merged we can update, but for now we have this!
```

Comment:
I'm wondering if it's worth it to deprecate using zenpy for new stuff in this repo anyway:

- Zenpy's documentation is really bad, so you have to basically reverse engineer how zenpy works by looking at code, trying stuff in a shell, and reading the zendesk api docs.
- Zenpy doesn't adopt new zendesk endpoints as fast as zendesk makes them, meaning we have to wait until stuff is supported and then upgrade zenpy, potentially breaking other things.

Seems like we could just cut out the middleman using two reasonably documented things (zendesk, which we can't avoid regardless, and requests).

Reply:
Agreed re: zenpy's terribleness, spent a lot of (too much) time spelunking through there. 

I started to go down the requests route (hence the leftover import) but then I couldn't quite figure out how to get the header information required for the request out of the Zenpy object. 

What/how do you think would be the best way to handle that? Could/would we pass through a new client to the `do_sync` function?